src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -1,30 +1,49 @@
 import * as React from 'react';
+import { observer } from 'mobx-react';
 import { ButtonGroup, DropdownButton, MenuItem, Toolbar } from 'patternfly-react';
 import { ButtonToolbar } from 'react-bootstrap';
 
 import { GraphFilterProps, GraphFilterState } from '../../types/GraphFilter';
-import * as API from '../../services/Api';
+import Namespace from '../../types/Namespace';
 import { DurationButtonGroup } from './DurationButtonGroup';
 import { LayoutButtonGroup } from './LayoutButtonGroup';
+import { namespaceStore } from '../../model/Namespace';
 
+type DropdownListType = {
```

Comment:
Just for my understanding: when is it better to use type, when is it better to use interface? Is it just a matter of taste, or is there any commonly accepted guidelines on this?

Reply:
I guess we should use `interfaces` when creating types that will be used externally to give more context when an error happens.

> One difference is that interfaces create a new name that is used everywhere. Type aliases don’t create a new name — for instance, error messages won’t use the alias name.

https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases

Reply:
To sum up, if you plan on inheriting or extending then `interface` is better. Type is better if its just local and/or you need to use `type guards`